cpp/src/arrow/compute/kernels/aggregate_basic_internal.h

```diff
@@ -289,27 +313,92 @@ struct MinMaxState<ArrowType, SimdLevel, enable_if_decimal<ArrowType>> {
     return *this;
   }
 
-  void MergeOne(const uint8_t* value) { MergeOne(T(value)); }
+  void MergeOne(util::string_view value) {
+    MergeOne(T(reinterpret_cast<const uint8_t*>(value.data())));
+  }
 
   void MergeOne(const T value) {
     this->min = std::min(this->min, value);
     this->max = std::max(this->max, value);
   }
 
+  Status GetValues(const std::shared_ptr<DataType>& type,
+                   std::vector<std::shared_ptr<Scalar>>* values) {
+    values->emplace_back(std::make_shared<ScalarType>(min, type));
+    values->emplace_back(std::make_shared<ScalarType>(max, type));
+    return Status::OK();
+  }
+
   T min;
   T max;
   bool has_nulls = false;
   bool has_values = false;
 };
 
+template <typename ArrowType, SimdLevel::type SimdLevel>
+struct MinMaxState<ArrowType, SimdLevel,
+                   enable_if_t<is_base_binary_type<ArrowType>::value ||
+                               std::is_same<ArrowType, FixedSizeBinaryType>::value>> {
+  using ThisType = MinMaxState<ArrowType, SimdLevel>;
+  using ScalarType = typename TypeTraits<ArrowType>::ScalarType;
+
+  ThisType& operator+=(const ThisType& rhs) {
+    if (!this->seen && rhs.seen) {
+      this->min = rhs.min;
+      this->max = rhs.max;
+    } else if (this->seen && rhs.seen) {
+      if (this->min > rhs.min) {
+        this->min = rhs.min;
+      }
+      if (this->max < rhs.max) {
+        this->max = rhs.max;
+      }
+    }
+    this->has_nulls |= rhs.has_nulls;
+    this->has_values |= rhs.has_values;
+    this->seen |= rhs.seen;
+    return *this;
+  }
+
+  void MergeOne(util::string_view value) {
+    if (!seen) {
+      this->min = std::string(value);
+      this->max = std::string(value);
+    } else {
+      if (value < util::string_view(this->min)) {
+        this->min = std::string(value);
+      }
+      if (value > util::string_view(this->max)) {
```

Comment:
Should this be `else if`? Presumably `value` cannot be bother smaller than the min and greater than the max, and it would spare a string comparison.